CMakeLists.txt

```diff
@@ -837,6 +842,31 @@ if(CONFIG_GEN_ISR_TABLES)
   set_property(GLOBAL APPEND PROPERTY GENERATED_KERNEL_SOURCE_FILES isr_tables.c)
 endif()
 
+if(CONFIG_ARM)
+add_custom_target(
+  ${CODE_DATA_RELOCATION}
+  DEPENDS app
+  kernel
```

Comment:
Can this break if a library other than 'app' or 'kernel' changes?

Reply:
can you please elaborate?

Reply:
There is no dependency from code_data_relocation to, for instance, subsys__bluetooth__controller.

So if a bluetooth file is modified, and this bluetooth file should be relocated by CODE_DATA_RELOCATION, then you will not re-generate CODE_DATA_RELOCATION, and you end up with a broken build.

You can test this by building. And then touching some source file that is not a part of 'app' or 'kernel', but that should be relocated. And then see if re-location is re-executed when "ninja/make" is invoked again (whithout cleaning the build directory).

Reply:
I understand. But how do i proceed?
Half the rules in our cmake are the same for generated files. 

We have been doing a fresh build since a long time if we expect the generated files to change.

IMHO I believe this comment is not pressing enough to stop the merge.

Reply:
It's true that there are some existing bugs related to this.

But nearly all C files can be modified without breaking the build.

This is not OK to merge.

Reply:
@SebastianBoe Done. please re-review.